src/ucp/api/ucp_def.h

```diff
@@ -625,9 +625,9 @@ typedef ucs_status_t (*ucp_am_callback_t)(void *arg, void *data, size_t length,
  *
  * @param [in]  arg           User-defined argument.
  * @param [in]  header        User defined active message header. Can be NULL.
```

Comment:
If I read it correctly, we don't need `Can be NULL`. Essentially the entire point of this change is to avoid reset to NUILL. Have we ever released this ?  If there are codes relaying on NULL check, it will break backward compatibility.

Reply:
The code itself does not set the header to NULL, so no break is expected

Reply:
Unless I'm missing something, the original code was setting the header to null if the lengths is null. Now this reset code is removed. if the CB application had a branch checking if the header is null (instead of length), it will cause for a segfault. I don't think we can call such code UCX compliant, nevertheless I think there is a chance for breaking app.

Reply:
adding the reset code was done after the release, so it should not affect applications using a released version

Reply:
Okay. Let's remove `can be NULL` statement